src/firefly/java/edu/caltech/ipac/firefly/server/util/QueryUtil.java

```diff
@@ -95,6 +96,38 @@ public static TableServerRequest convertToServerRequest(HttpServletRequest req)
         return retval;
     }
 
+    public static TableServerRequest convertToServerRequest(String str) {
```

Comment:
I am glad you ask.  QueryUtil uses URLDecoder.decode to decode nested parameter values.  TableServerRequest is used on both the client by gwt and the server. gwt cannot use URLDecoder.  Therefore, a simple decoder function is used there.  When we no longer need to support gwt, I plan to move the parsing logic back into TableServerRequest.